pkg/controller/appconfiguration/keyvalue.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	"github.com/stone-payments/appconfig-go-sdk/appconfig/keyvalues"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/crossplane/provider-azure/apis/appconfiguration/v1alpha1"
+	azure "github.com/crossplane/provider-azure/pkg/clients"
+	appclient "github.com/crossplane/provider-azure/pkg/clients/appconfiguration"
+)
+
+const (
+	errConnectFailed         = "cannot connect to Azure API"
+	errGetFailed             = "cannot get KeyValue"
+	errCreateFailed          = "cannot create KeyValue"
+	errUpdateFailed          = "cannot update KeyValue"
+	errKubeUpdateFailed      = "cannot update custom resource in kubernetes"
+	errDeleteFailed          = "cannot delete KeyValue"
+	errCannotCreateNewClient = "cannot create NewClient"
+	errNotKeyValue           = "unexpected resource, it must be KeyValue"
+)
+
+// SetupAppConfiguration adds a controller that reconciles KeyValue resources.
+func SetupAppConfiguration(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(v1alpha1.KeyValueKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.KeyValue{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.KeyValueGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	content, _, err := azure.GetAuthInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, errConnectFailed)
+	}
+
+	cr, ok := mg.(*v1alpha1.KeyValue)
+	if !ok {
+		return &external{}, errors.New(errNotKeyValue)
+	}
+
+	clientArgs := keyvalues.NewClientAzureADArgs{
+		ClientID:         content["clientId"],
+		ClientSecret:     content["clientSecret"],
+		TenantID:         content["tenantId"],
+		AADEndpoint:      content["activeDirectoryEndpointUrl"],
+		ResourceEndpoint: cr.Spec.ForProvider.Endpoint,
+	}
+
+	cl, err := keyvalues.NewClientAzureAD(clientArgs)
+	if err != nil {
+		return nil, errors.Wrap(err, errCannotCreateNewClient)
+	}
+	return &external{kube: c.kube, client: cl}, nil
```

Comment:
You can to use the auth returned from GetAuthInfo to generate new client without recall Authorizer
```suggestion
	_, auth, err := azure.GetAuthInfo(ctx, c.kube, mg)
	if err != nil {
		return nil, errors.Wrap(err, errConnectFailed)
	}

	cr, ok := mg.(*v1alpha1.KeyValue)
	if !ok {
		return &external{}, errors.New(errNotKeyValue)
	}

	cl := keyvalues.NewClient(cr.Spec.ForProvider.Endpoint, auth)

	return &external{kube: c.kube, client: cl}, nil
```

Reply:
This solution doesn't work